src/AutoRest.CSharp.V3/Generation/Writers/JsonCodeWriterExtensions.cs

```diff
@@ -227,10 +227,11 @@ void WriteNullCheck()
                             if (property.Property != null)
                             {
                                 // Reading a property value
-                                writer.DeserializeIntoVariable(
+                                writer.DeserializeIntoVariableWithNullHandling(
```

Comment:
You don't need to call DeserializeIntoVariableWithNullHandling here if you are passing false anyway.

Reply:
Done.